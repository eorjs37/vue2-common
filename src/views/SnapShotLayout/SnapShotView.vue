<template>
	<div>
		<div id="admin" v-if="ROLE === 'ADMIN'">
			관리자
		</div>
		<div id="admin" v-else-if="ROLE === 'MEMBER'">
			멤버1
		</div>
		<div id="admin" v-else-if="ROLE === 'GUEST'">
			게스트
		</div>
	</div>
</template>
<script setup>
import { ref } from 'vue';


const ROLE = ref('ADMIN')

</script>