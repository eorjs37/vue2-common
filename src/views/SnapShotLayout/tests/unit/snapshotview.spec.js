import { shallowMount } from '@vue/test-utils';

import SnapShotView from '../../SnapShotView.vue';

describe('snapshotview unit testing', () => {
	let wrapper = null;
	beforeEach(() => {
		wrapper = shallowMount(SnapShotView)
	});
	afterEach(() => {
		wrapper.vm.ROLE = 'GUEST'
	});
	test('관리자일때 화면 테스트', async () => {
		//given
		wrapper.vm.ROLE = 'ADMIN'

		//when

		//then
		expect(wrapper.vm.$el).toMatchSnapshot()

	});

	test('멤버일때 화면 테스트', async () => {
		//given
		wrapper.vm.ROLE = 'MEMBER'

		//when
		await wrapper.vm.$nextTick();
		//then
		expect(wrapper.vm.$el).toMatchSnapshot()

	});

	test('게스트일때 화면 테스트', async () => {
		//given
		wrapper.vm.ROLE = 'GUEST'

		//when
		await wrapper.vm.$nextTick();
		//then
		expect(wrapper.vm.$el).toMatchSnapshot()

	});
});