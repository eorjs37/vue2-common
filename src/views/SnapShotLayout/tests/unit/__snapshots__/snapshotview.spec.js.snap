// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshotview unit testing 게스트일때 화면 테스트 1`] = `
<div>
  <div
    id="admin"
  >
     게스트 
  </div>
</div>
`;

exports[`snapshotview unit testing 관리자일때 화면 테스트 1`] = `
<div>
  <div
    id="admin"
  >
     관리자 
  </div>
</div>
`;

exports[`snapshotview unit testing 멤버일때 화면 테스트 1`] = `
<div>
  <div
    id="admin"
  >
     멤버1 
  </div>
</div>
`;
